src/services/expenses.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+class Expenses {
+  constructor() {
+    this.expenses = [];
+  }
+
+  create(data) {
+    const maxId = Math.max(...this.expenses.map(user => user.id));
+    const id = (this.expenses.length > 0) ? maxId + 1 : 1;
+
+    const newExpense = {
+      id,
+      ...data,
+    };
+
+    this.expenses.push(newExpense);
+
+    return newExpense;
+  }
+
+  getAll(userId, category, from, to) {
+    const filteredExpenses = this.expenses.filter(expense => {
+      const { spentAt } = expense;
+
+      if (userId && expense.userId !== +userId) {
+        return false;
+      }
+
+      if ((from && to) && (spentAt.localeCompare(from) === -1
+      || spentAt.localeCompare(to) === 1)) {
+        return false;
+      }
+
+      if (category && expense.category !== category) {
+        return false;
+      }
```

Comment:
You need to check an array, not a single value.
![Accounting-App-1-0-0-YURIIHOLIUK_1-SwaggerHub](https://user-images.githubusercontent.com/54940595/204246720-e068a021-b4ec-43ab-957e-aef7fe90e42e.png)